test/mock_transport_test.exs

```diff
@@ -0,0 +1,93 @@
+defmodule Janus.MockTransportTest do
+  use ExUnit.Case
+
+  alias Janus.MockTransport
+
+  @request_response_pairs [
+    {
+      %{
+        janus: :create
+      },
+      %{
+        "janus" => "success",
+        "data" => %{"id" => "session id"}
+      }
+    },
+    {
+      %{
+        janus: :keepalive
+      },
+      %{
+        "janus" => "ack"
+      }
+    },
+    {
+      %{
+        janus: :keepalive
+      },
+      %{
+        "janus" => "ack"
+      }
+    }
+  ]
+
+  describe "MockTransport should" do
```

Comment:
I would add a test that specificaly checks that if request is present only once in the the provided state only the first try results in success, and the second fails.